src/main/java/com/github/maltalex/ineter/range/IPRange.java

```diff
@@ -9,17 +9,21 @@
 
 import java.io.Serializable;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
 
 import com.github.maltalex.ineter.base.IPAddress;
 
-public interface IPRange<I extends IPAddress & Comparable<I>, L extends Number & Comparable<L>>
-		extends Iterable<I>, Serializable {
+public interface IPRange<
```

Comment:
That's interesting.

I believe I've already been down this road at some point with this project, although I'm not sure I got to as many generic bindings as you have.
I gave up on my attempt because trying to write generic classes on top of the generic "IPRange" turned out to be a mess, but maybe what was needed was **more** generics, like you have done, not less.

I like that it allows the interface to expose more methods, but I'm afraid to imagine what generic code that uses this class would look like. I'll merge it in and try to build something on top, see how it goes.

By the way, `IPAddress` could have also been generified to allow for methods like `next` and `compareTo` **in the interface** to accept/return an instance of the same type of address. But I like it that you stopped with the ranges/subnets. Users should be use `IPAddress` comfortably without generics.